app/src/org/commcare/android/database/DbHelper.java

```diff
@@ -103,6 +103,10 @@ public ContentValues getContentValues(Externalizable e) {
                 values.put(TableBuilder.scrubName(key), value);
             }
         }
+
+        if(blob.length > Math.pow(1024, 2)){
```

Comment:
I can't remember the exact path it used, but it was thrown in such an obscure way that it was very difficult to catch the actual exception at an actionable point. Considering we'll want to refactor this to use the file system at some point I didn't want to sink too much into this fix